src/utils.rs

```diff
@@ -8,10 +8,24 @@ use tilejson::Bounds;
 
 use crate::source::{Query, Xyz};
 
-pub fn prettify_error<E: std::fmt::Display>(message: String) -> impl Fn(E) -> std::io::Error {
-    move |error| std::io::Error::new(std::io::ErrorKind::Other, format!("{message}: {error}"))
+#[macro_export]
+macro_rules! prettify_error {
+    ($error:ident, $info:expr, $($arg:expr),+ $(,)?) => {
+        {
+            ::std::io::Error::new(::std::io::ErrorKind::Other, format!("{}: {}",
+            format!($info, $($arg),*),
+            $error))
+        }
+    };
+    ($error:ident, $info:literal) => {
+        {
+            ::std::io::Error::new(::std::io::ErrorKind::Other, format!(concat!($info, ": {}"), $error))
+        }
+    }
```

Comment:
I think this will be more accurate, and it will consume the literal variant first.  I did some digging into the exports and macros -- turns out there is a hidden "bug" we have in the code -- the macro is actually exported twice -- once as a top level public function, and another as a macro inside the utils module, but only for internal usage.  Not a big deal of course, but it did cause some confusion for me :)

Regardless, I think we can even merge it as is, or if you accept my proposed change, we can merge it with it.

```suggestion
    ($error:ident, $info:literal) => {
        ::std::io::Error::new(::std::io::ErrorKind::Other, format!(concat!($info, ": {}"), $error))
    };
    ($error:ident, $($arg:tt)+) => {
        ::std::io::Error::new(::std::io::ErrorKind::Other,
            format!("{}: {}", format_args!($($arg)+), $error))
    };
```